[vcpkg] Expand Architecture list with escape chars

sandercox opened the pull request:
**Describe the pull request**
This expands the architecture lists with escape characters. When
building FAT binaries for macos using multiple architectures in the
values they need to be escaped otherwise they are passed on to CMake
incorrectly #14932

- #### What does your PR fix?  
  Fixes #14932

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `yes`

sandercox committed:
Expand Architecture list with escape chars

This expands the architecture lists with escape characters. When
building FAT binaries for macos using multiple architectures in the
values they need to be escaped otherwise they are passed on to CMake
incorrectly #14932

NancyLi1013 commented:
```
CMake Error at src/tm2/CMakeLists.txt:16 (message):
  BUILD_WITH_TM2 requires IMPORT_DEPTH_CAM_FW
Call Stack (most recent call first):
  src/CMakeLists.txt:57 (include)
  CMakeLists.txt:56 (include)
```

The failure caused by `realsense2 `on x64-osx will be fixed by PR #18202.

NancyLi1013 commented:
/azp run

strega-nil-ms commented:
Could you fix this same bug in `vcpkg_cmake_configure`? Additionally, it's likely we'll want to set up a better way to do this, similar to `vcpkg_function_arguments`.

sandercox commented:
@strega-nil-ms thanks for pointing out the vcpkg_cmake_configure. Was having issues with fmt lib and fixing that file fixed the issue.

Fixed that and rebased on master as there were other changes in the vcpkg-cmake port file

omartijn commented:
/azp run

NancyLi1013 commented:
/azp run

sandercox committed:
Adding the architecture fix to vcpkg-cmake port
updating port vcpkg-cmake version

sandercox commented:
There was a small typo in the attempted fix for `vcpkg_cmake_configure` also rebased on master again.

NancyLi1013 reviewed (commented):
This file is also needed to update.
https://github.com/microsoft/vcpkg/blob/master/versions/baseline.json

sandercox committed:
updated version in baseline

sandercox commented:
@NancyLi1013 updated `versions/baseline.json` too. Thanks for pointing that out.

ras0219-msft commented:
This LGTM, thanks for the fix!
